org-ql.el

```diff
@@ -448,6 +452,29 @@ Returns cons (INHERITED-TAGS . LOCAL-TAGS)."
                  org-ql-tags-cache))
       (puthash position all-tags tags-cache))))
 
+(defun org-ql--expand-tag-groups (tags &optional exclude-groups)
+  (if (eq tags 'org-ql-nil)
+      'org-ql-nil
+    (let ((exclude-groups (append tags exclude-groups))
+          result)
+      (let (group-tags)
+        (dolist (tag tags)
+          (dolist (groups org-tag-groups-alist)
+            (when (and (not (cl-member (car groups) exclude-groups
```

Comment:
It would be very helpful to write a comment explaining what these tests do in plain English.  Not having used the feature myself, and not having stepped through the code, I don't understand what the regexp tests are for or why they're necessary.

Reply:
I honestly have no excuse here. According to the manual, the members of a tag group can be either tags to be matched exactly or regular expressions, which are denoted by enclosing them in curly brackets. Addressed in the new commit.